mutation-testing/output-pr-mutants/action.yml

```
@@ -0,0 +1,166 @@
+name: Output Mutants
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Download artifacts
+      uses: actions/download-artifact@v3
+
+    - name: Append output from shards
+      shell: bash
+      run: |
+        # List of all possible mutant outcomes file names and folder names collected from jobs.
+        folders=("mutants-shard-big--1" "mutants-shard-big-0" "mutants-shard-big-1" "mutants-shard-big-2" "mutants-shard-big-3" "mutants-shard-big-4" "mutants-shard-big-5" "mutants-shard-big-6" "mutants-shard-big-7" "mutants-shard-small--1" "mutants-shard-small-0" "mutants-shard-small-1" "mutants-shard-small-2" "mutants-shard-small-3")
```

i think the first folder name here has an extra `-`. 

I'm also curious why these folder names are hardcoded - are these folders based on the the matrix of the calling workflow? if that is true, i think we can create a loop to do this step vs hardcoding the names. 


```suggestion
        folders=("mutants-shard-big-1" "mutants-shard-big-0" "mutants-shard-big-1" "mutants-shard-big-2" "mutants-shard-big-3" "mutants-shard-big-4" "mutants-shard-big-5" "mutants-shard-big-6" "mutants-shard-big-7" "mutants-shard-small--1" "mutants-shard-small-0" "mutants-shard-small-1" "mutants-shard-small-2" "mutants-shard-small-3")
```

---

ultimately, this is *fine* but i think it can/should be cleaner if possible.

---

'-1' is the default input for running without shards, hence the doubled dashes (https://github.com/stacks-network/actions/blob/feat/mutation-testing/mutation-testing/pr-differences/action.yml#L4-L7). 'mutants-shard-big-1' already exists and it's part of the matrix of shards.

The folders are hardcoded for the matrix jobs, how can i get the matrixes length in the composite action in order to use that directly in the 'for' loop?

---

you can do that via an input!
not a blocker for this PR, but perhaps a future improvement. 

say you have a calling workflow that uses a matrix of 8 jobs. 
when you call this composite workflow, you can send the number of jobs as an input:
```
uses: this workflow
with: 
  jobs: 8 
```
and then, in `output-pr-mutants/action.yml` you simple add an input with a sane default (8). 
in the shell, you can then do a simple for loop, i.e (pseudocode).:
```bash
for i in {1..${{inputs.matrix-jobs }}; do
  mkdir -p mutants-shard-big-$i
done
```

ref: https://github.com/orgs/community/discussions/26930#discussioncomment-3254002

i think i'm actually hardcoding this value myself in at least one place - so i'l have to test if this method works as i expect.

---

As we've discussed, I did it by parsing the type of operations it will do and that will know the number of shards it should have. This way, it will generate the specific folders, and only those will be tested in the workflows.